internal/upgradecheck/helpers_test.go

```diff
@@ -0,0 +1,154 @@
+package upgradecheck
```

Comment:
Moving certain helpers from `header_test.go` to here so that they can also be used by `http_test.go`